pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1408,9 +1408,11 @@ void sendComplete(final Exception e) {
                         te.getMessage(),
                         RelativeTimeUtil.nsToSeconds(ns - this.createdAt),
                         RelativeTimeUtil.nsToSeconds(this.firstSentAt <= 0
-                                ? ns - this.lastSentAt
```

Comment:
It is interesting. Do you mean setting it to -1 as `firstSendAt` value if `firstSendAt <= 0`?

Reply:
yep. -1 represents the msg not flush.